lib/Dialect/Torch/Transforms/ReduceOpVariants.cpp

```diff
@@ -39,68 +39,81 @@ class ConvertToImmutableTensors : public RewritePattern {
         opOperand.set(rewriter.create<CopyToValueTensorOp>(op->getLoc(),
                                                            opOperand.get()));
       } else if (auto listType = operandType.dyn_cast<ListType>()) {
-        if (!listType.getContainedType().isa<NonValueTensorType>())
-          continue;
-
-        // Construct a new list whose elements are value tensors copied from
-        // the none value tensors of the original list.
-        auto listConstruct =
-            opOperand.get().getDefiningOp<PrimListConstructOp>();
-        if (!listConstruct) {
-          rewriter.cancelRootUpdate(op);
-          return rewriter.notifyMatchFailure(op, 
-              "unimplemented: list of non vtensor type not constructed "
-              "from list construct");
-        }
+        if (!(listType.getContainedType().isa<NonValueTensorType>() ||
+              listType.getContainedType().isa<OptionalType>()))
+            continue;
```

Comment:
I think these are the same changes that @pashu123 added in https://github.com/llvm/torch-mlir/pull/488, so I'm copying over the comments to here:

https://github.com/llvm/torch-mlir/pull/488#discussion_r779109555
https://github.com/llvm/torch-mlir/pull/488#discussion_r779176882

An example of how the formatting is off is the `continue;` above. It should be indented two spaces from the starting column of the `if`.